pageserver/src/remote_storage/storage_sync/download.rs

```diff
@@ -149,6 +158,19 @@ where
                             "Failed to download a layer from storage path '{layer_storage_path:?}'"
                         )
                     })?;
+                destination_file.sync_data().await?;
+
+                // fsync parent dir
+                let parent_path = temp_file_path
+                    .parent()
+                    .expect("no parent dir for layer file");
+                let parent_dir = fs::File::open(parent_path).await?;
+                parent_dir.sync_all().await?;
+
+                fail::fail_point!("remote-storage-download-pre-rename", |_| {
+                    bail!("failpoint triggered")
+                });
+                fs::rename(temp_file_path, &layer_desination_path).await?;
```

Comment:
I wonder, should we try fsyncing those in parallel after all download sync tasks are done, with par_fsync that we have?

Also, this `fs::rename`, should it be also somehow fsynced? 

Last but not least, we do sync `metadata` lately, one level above, in `update_local_metadata`.
Should we do both/one of/neither `fsync` and `*.tmp` move for it too?